cmd/agent/subcommands/run/command.go

```diff
@@ -181,6 +182,7 @@ func run(log log.Component,
 	sharedSerializer serializer.MetricSerializer,
 	cliParams *cliParams,
 	logsAgent util.Optional[logsAgent.Component],
+	_ server.Component,
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
**❓ question:** Curious, why this is needed?‏

Reply:
Adding the bundle to `getSharedFxOption` tells Fx how to create a new server, but that doesn't actually cause it create or run one - it only gets created if `fx.Invoke` is called on a function that requires it. Adding it here as a dummy argument means Fx will create the server in order to run this function, even though the function doesn't actually use it.

We could get the same effect by adding our own `Invoke` block that relies on the server, but this breaks the current subcommand tests because it means that even if the `run` function isn't actually run we'll still try to initialize our service.